File: SKLibrary/src/main/java/com/silvericekey/skutilslibrary/SKUtilsLibrary.java
package com.silvericekey.skutilslibrary;

import android.content.Context;
import android.util.DisplayMetrics;

import com.facebook.drawee.backends.pipeline.Fresco;

import me.jessyan.autosize.AutoSize;
import me.jessyan.autosize.AutoSizeConfig;
import me.jessyan.autosize.unit.Subunits;
import me.jessyan.autosize.unit.UnitsManager;

/**
 * Created by silverknife on 2017/12/13.
 */

public class SKUtilsLibrary {
    private static Context mContext;

    public static void init(Context context){
        mContext = context;
        Fresco.initialize(context);
    }

    public static Context getContext() {
        if (mContext==null){
            throw new NullPointerException("Please init SKUtilsLibrary in your Application or other place before you use");
        }
        return mContext;
    }
}

File: SKLibrary/src/main/java/com/silvericekey/skutilslibrary/SKUtilsLibrary.kt
package com.silvericekey.skutilslibrary

import android.annotation.SuppressLint
import android.content.Context
import com.blankj.utilcode.util.Utils
import com.facebook.drawee.backends.pipeline.Fresco


/**
 * Created by silverknife on 2017/12/13.
 */

@SuppressLint("StaticFieldLeak")
object SKUtilsLibrary {

    /**
     * 获取Context如果没有初始化会抛出异常
     */
    @JvmStatic
    var context: Context? = null
        get() {
            if (field == null) {
                throw NullPointerException("please init SKUtilsLibrary first")
            }
            return field
        }

    /**
     * 框架初始化，获取ApplicationContext
     * */
    fun init(context: Context) {
        this.context = context
        Utils.init(context)
        Fresco.initialize(context)
    }
}
